extras/README.md

```diff
@@ -1,16 +1,97 @@
-# New Actor Template
-Use cargo generate to create a new actor module from this template.
+## How to use  built-in `wasmcloud:extras` capability provider
 
-[Learn how to write](https://wascc.dev/tutorials/first-actor/) waSCC actors, or [explore the actor concept](https://wascc.dev/docs/concepts/actors/).
+`wasmcloud` host runtime comes with some built-in capability providers and `wasmcloud:extras` is one of them. In this tutorial we will generate a random number and retrive some information like request id and number of requests using `wasmcloud:extras`. 
 
-To generate new keys, use `make keys`. This will use `nk` to generate both sets of keys for you, and then write them to the `.keys` directory.
+1. Please install `wasmcloud` and `wash` binaries if you are not already did:
 
-To build your new module, use `make build`. This will compile your code with `cargo`, and then sign it with `wascap` using the keys in `.keys`.
+    https://wasmcloud.dev/overview/installation/
 
-## Tool Requirements
+    ```bash
+    $ wasmcloud --version
+    $ wash --version
+    ```
 
-- Cargo and Rust are required
-- Make is recommended, but not strictly necessary
-- [wascap](https://github.com/wascc/wascap) is required
-- [nk](https://github.com/encabulators/nkeys) is required if you need to generate keys (which you almost certainly do)
+2. Compile and sign the actor:
 
+    ```bash
+    $ cargo build --release
+
+    $ wash claims sign ./target/wasm32-unknown-unknown/release/extras.wasm -c wasmcloud:httpserver -c c wasmcloud:extras --name "extras" --ver 0.2.1
+    ```
+
+    Please note we are signing our actor for multiple capabilities including built-in ones: `wasmcloud:httpserver` and `wasmcloud:extras`. Failing to do so causes mulfunctioning of the actor.
+
+3. Get actor id:
+
+    ```bash
+    $ wash claims inspect ./target/wasm32-unknown-unknown/release/extras_s.wasm
+    ```
+
+    This will output cryptographically signed id for our actor like:
+
+    ```bash
+     Module       MDAR7HG42IAZBHJQNS2CQPYJPCSN55CDCKBYSUDVY2R4ZB6YDFONID2B 
+    ```
+
+4. Now it is time to get provider id for `wasmcloud:extras`:
+
+    ```
+    // Start a REPL session
+    $ wash up
+
+    // Get host id
+    $ ctl get hosts
+
+    // Retrieve inventory detail using host id
+    $ wash ctl get inventory «HOST_ID»
+    ```
+
+    This will output something like:
+
+    ```bash
+    Provider ID                                                Link Name                  Image Reference
+    VAHNM37G4ARHZ3CYHB3L34M6TYQWQR6IZ4QVYC4NYZWTJCJ2LWP7S6Z2   __wasmcloud_lattice_cache  N/A
+    VDHPKGFKDI34Y4RN4PWWZHRYZ6373HYRSNNEM4UTDLLOGO5B37TSVREP   default                    N/A          
+    ```
+
+    We need the id for the `default` one. We are using remote capability provider for `wasmcloud:httpserver` and it's id is provided below.
```

Comment:
"it's" can just be "its" here 🙂

Reply:
Also after the sentence where you say "We need the id for the `default` one", can you also put in a sentence that says "We are in the process of adding the provider name into the inventory output."